design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,187 @@
+
+1. The user creates an `ApplicationStackRequest` and specifies the WordPress stack
+1. The Stack Manager installs the CRDs defined in the stack, creates the necessary RBAC primitives, and starts up the WordPress controller in the same namespace as the stack.
+1. The user creates an instance of the `WordPressInstance` custom resource in the same namespace.
+1. The WordPress controller is watching for events on this CRD type in the namespace it is running in.
+1. In response to seeing the creation event for this `WordPressInstance` custom resource, the WordPress controller creates claims for the services that the instance will need, such as a MySQL database and a Kubernetes cluster.
+1. These claims are then serviced by the default resource classes in the environment that point to specific provisioners that were previously installed as part of a platform stack.
+
+This flow demonstrates how an application stack indirectly consumes a platform stack and is represented in the diagram below:
+
+![Diagram of an Application Stack Installed](./images/application-stack-installed.png)
+
+### Multiple Versions
+
+As previously mentioned, since application stacks will be installed per namespace, multiple instances of each stack can be installed in the control plane.
+This also means that multiple versions of the stack can be installed side by side.
+Because of the namespaced isolation, this is fine for the stack's controllers, but it does cause a potential difficulty for the CRDs.
+
+The custom resource **definitions** themselves are always cluster scoped.
+If two versions of an application stack are installed in the control plane and they have identical CRD versions, there could be a conflict.
+To illustrate this clearly, consider the following example:
+
+1. App stack `foo-v1.0` is installed along with its CRD that has version `v1beta1`
+1. A minor update to app stack foo is released as version `foo-v1.1`
+    1. This update to the stack adds a new field to the CRD but does **not** increment the CRD version
+1. The control plane now has 2 versions of the stack's controllers (`foo-v1.0` and `foo-v1.1`), but only 1 version of the CRD (`v1beta`)
+
+Which CRD schema should "win" in the control plane?
+
+* The **new CRD schema** with the new field that the **old controller** doesn't understand?
+* The **old CRD schema** without the new field that the **new controller** wants to use?
+
+This speaks to the importance of properly versioning CRDs when they have schema changes.
+Kubernetes now supports [multiple side by side versions of CRDs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions) in the cluster, so this is actually possible in more recent versions of Kubernetes.
+We suspect (and hope) that this newer CRD versions functionality will encourage CRD authors to adopt a more strict process of updating their CRD versions, which would avoid the issue raised in this section.
```

Comment:
I think it is safe to make the assumption that stack developers will be cognizant of the fact that they should increment CRD version on changes. However, we probably will want to specify this in any guides / tooling we create to support them.